src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -70,14 +70,16 @@ public class TxPool : ITxPool, IDisposable
         /// <param name="logManager"></param>
         /// <param name="comparer"></param>
         /// <param name="incomingTxFilter"></param>
+        /// <param name="thereIsPriorityContract"></param>
         public TxPool(
             IEthereumEcdsa ecdsa,
             IChainHeadInfoProvider chainHeadInfoProvider,
             ITxPoolConfig txPoolConfig,
             ITxValidator validator,
             ILogManager? logManager,
             IComparer<Transaction> comparer,
-            IIncomingTxFilter? incomingTxFilter = null)
+            IIncomingTxFilter? incomingTxFilter = null,
```

Comment:
Why not use IIncomingTxFilter here for non-aura to pass those filters only when needed?

Reply:
Wrong points in chain (2 different points); and filters are trying to reduce state access but incomingTxnFilter both occurs after the main state access and currently doesn't have access to required data; need headinfo, and txn queue